### PR TITLE
Swap samtools collatefastq to nf-test

### PR DESCRIPTION
**Pull request by SPPearce**
Swapping samtools collatefastq to nf-test.
I also found that the pytests weren't actually using the files that they claimed to be, and that the interleaved function was actually producing a fastq file, not the fastq.gz file it claimed to be.

### CODE REVIEW

**Comment by SPPearce** [comment]
Currently the interleaved option makes a fastq file, rather than a fastq.gz file.  
I have adjusted the name and the emit statement to match this, but probably wants to have a conditional gzip in the process to zip it up.

File: modules/nf-core/samtools/collatefastq/tests/nextflow.config
Comment (maxulysse) [inline modules/nf-core/samtools/collatefastq/tests/nextflow.config:1]: I think this file can be deleted

**Comment by maxulysse** [comment]
minor comments

File: modules/nf-core/samtools/collatefastq/main.nf [hunk 1]
```diff
@@ -52,4 +52,20 @@ process SAMTOOLS_COLLATEFASTQ {
         samtools: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//')
     END_VERSIONS
     """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def outputcommand =    (interleave && ! meta.single_end) ? "echo '' | gzip > ${prefix}_interleaved.fq.gz"                     :
+                    meta.single_end                   ? "echo '' | gzip > ${prefix}_1.fq.gz ; echo '' | gzip > ${prefix}_singleton.fq.gz" :
+                    "echo '' | gzip > ${prefix}_1.fq.gz ; echo '' | gzip > ${prefix}_2.fq.gz ; echo '' | gzip > ${prefix}_singleton.fq.gz"
```
Comment (nvnieuwk) [inline modules/nf-core/samtools/collatefastq/main.nf:60]: Can you make this a bit more readable? :grin:

File: modules/nf-core/samtools/collatefastq/tests/main.nf.test.snap [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        "content": [
+            [
+                
+            ],
+            [
+                
+            ],
+            [
+                
+            ],
```
Comment (nvnieuwk) [inline modules/nf-core/samtools/collatefastq/tests/main.nf.test.snap:12]: The stub doesn't output anything, this doesn't seem right?

**Comment by SPPearce** [comment]
@nvnieuwk, I have cleaned up the stub if you want to take a look.

**Review by nvnieuwk — APPROVED** [review]
Much better!

**Pull request merged**